lib/temporal/connection/grpc.rb

```diff
@@ -174,12 +174,41 @@ def poll_workflow_task_queue(namespace:, task_queue:)
         poll_request.execute
       end
 
-      def respond_workflow_task_completed(namespace:, task_token:, commands:)
+      def respond_query_task_completed(namespace:, task_token:, query_result: nil, error_message: nil)
+        request = Temporal::Api::WorkflowService::V1::RespondQueryTaskCompletedRequest.new(
+          task_token: task_token,
+          namespace: namespace
+        )
+        if !error_message.nil?
+          request.error_message = error_message
+          request.completed_type = Temporal::Api::Enums::V1::QueryResultType::QUERY_RESULT_TYPE_FAILED
+        else
+          request.query_result = to_query_payloads(query_result)
+          request.completed_type = Temporal::Api::Enums::V1::QueryResultType::QUERY_RESULT_TYPE_ANSWERED
+        end
+        client.respond_query_task_completed(request)
+      end
+
+      def respond_workflow_task_completed(namespace:, task_token:, commands:, query_results:)
+        query_results&.transform_values! do |query_result|
+          if query_result.is_a?(StandardError)
+            {
+              result_type: Temporal::Api::Enums::V1::QueryResultType::QUERY_RESULT_TYPE_FAILED,
+              error_message: query_result.message
+            }
+          else
+            {
+              result_type: Temporal::Api::Enums::V1::QueryResultType::QUERY_RESULT_TYPE_ANSWERED,
+              answer: to_query_payloads(query_result)
+            }
+          end
+        end
```

Comment:
I don't love the approach here of passing through exception objects as query results, happy to discuss alternatives